os/os-mac.h

```diff
@@ -117,3 +123,58 @@ static inline bool os_cpu_has(cpu_features feature)
 }
 
 #endif
+
+/*
+ * discard_pages should not be run within Rosetta. Native arm64 or x86 only. 
+ */
+
+#define MMAP_CHUNK_SIZE		(1024 * 1024 * 1024)
+
+static inline int discard_pages(int fd, off_t offset, off_t size)
+{
+	caddr_t *addr;
+	uint64_t chunk_size = MMAP_CHUNK_SIZE;
+
+	if (fsync(fd) < 0) {
```

Comment:
Why do you `fsync()` here? Is it to speed up the invalidation in the case that there's cached data that need flushing to disk? If you want flushing semantics is this better than using the `MS_SYNC` flag?

I'm even wondering whether we need to do any flushing at all...